lib/byron/test/integration/Cardano/Wallet/Byron/Config.hs

```diff
@@ -0,0 +1,251 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.Wallet.Byron.Config
+    ( -- * Integration Launcher
+      withCardanoNode
+    ) where
+
+import Prelude
+
+import Cardano.BM.Trace
+    ( Trace )
+import Cardano.Chain.Genesis
+    ( GenesisData (..), readGenesisData )
+import Cardano.Crypto.ProtocolMagic
+    ( ProtocolMagicId (..) )
+import Cardano.Launcher
+    ( Command (..), StdStream (..), withBackendProcess )
+import Cardano.Wallet.Byron.Compatibility
+    ( NodeVersionData, fromGenesisData )
+import Cardano.Wallet.Byron.Network
+    ( AddrInfo, localSocketAddrInfo )
+import Cardano.Wallet.Byron.Transaction
+    ( fromGenesisTxOut )
+import Cardano.Wallet.Logging
+    ( trMessageText )
+import Cardano.Wallet.Network.Ports
+    ( getRandomPort )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( hex )
+import Cardano.Wallet.Primitive.Types
+    ( Block (..)
+    , BlockHeader (..)
+    , BlockchainParameters (..)
+    , Hash (..)
+    , SlotId (..)
+    , TxOut (..)
+    )
+import Cardano.Wallet.Unsafe
+    ( unsafeRunExceptT )
+import Control.Exception
+    ( bracket, throwIO )
+import Control.Monad
+    ( forM_, unless, when )
+import Control.Monad.Fail
+    ( MonadFail )
+import Data.Aeson
+    ( toJSON )
+import Data.Coerce
+    ( coerce )
+import Data.Quantity
+    ( Quantity (..) )
+import Data.Text
+    ( Text )
+import Data.Time.Clock.POSIX
+    ( getPOSIXTime )
+import Ouroboros.Network.Magic
+    ( NetworkMagic (..) )
+import Ouroboros.Network.NodeToClient
+    ( NodeToClientVersionData (..), nodeToClientCodecCBORTerm )
+import System.Directory
+    ( copyFile, doesDirectoryExist, removeDirectoryRecursive )
+import System.Environment
+    ( lookupEnv )
+import System.FilePath
+    ( (</>) )
+import System.IO.Temp
+    ( createTempDirectory, getCanonicalTemporaryDirectory )
+import Test.Utils.Paths
+    ( getTestData )
+
+import qualified Data.Aeson as Aeson
+import qualified Data.ByteString as BS
+import qualified Data.HashMap.Strict as HM
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import qualified Data.Yaml as Yaml
+
+data CardanoNodeConfig = CardanoNodeConfig
+    { nodeGenesisHash  :: Text
+    , nodeConfigFile   :: FilePath
+    , nodeDatabaseDir  :: FilePath
+    , nodeDlgCertFile  :: FilePath
+    , nodeGenesisFile  :: FilePath
+    , nodeSignKeyFile  :: FilePath
+    , nodeSocketFile   :: FilePath
+    , nodeTopologyFile :: FilePath
+    }
+
+-- | Spins up a @cardano-node@ in another process.
+--
+-- IMPORTANT: @cardano-node@ must be available on the current path.
+withCardanoNode
+    :: Trace IO Text
+    -- ^ Some trace for logging
+    -> (AddrInfo -> Block -> (BlockchainParameters, NodeVersionData) -> IO a)
+    -- ^ Callback function with a socket description and genesis params
+    -> IO a
+withCardanoNode tr action =
```

Comment:
if `cardano-node` is not on current path are we going to get adequate msg?

Reply:
Good question :thinking: I'll check that.

Reply:
```
  1) API Specifications, BYRON_TXS, TRANS_CREATE_01 - Single Output Transaction
       uncaught exception: ProcessHasExited
       ProcessDidNotStart "cardano-patate" cardano-patate: createProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
```